### ISSUE TITLE
Unexpected syntax errors of `a < (b >= c ...)`  in TS 4.7

### ISSUE THREAD

**Issue by gdh1995** [microsoft/TypeScript#49551]
# Bug Report

### 🔎 Search Terms

- Syntax error
- error TS1005

### 🕗 Version & Regression Information

- This changed between versions __4.6.3__ and __4.7.2__

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/EQVwzgpgBGAuBOBLAxrYBuAsAKGQewDs4oICQBbKAWQE8A1CeKAbygDcBGKAXig4Bp2AJh5QRAXxwAbCLHaNRQjjnxE5AQ1FsFAHigAKWg3gA6TlAB8vI4zMiA-NXq3zALifG7ASiA)

### 💻 Code

```ts
"use strict";
const enum MyVer { v1 = 1, v2 = 2 }
let ver = 21
const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
```

### 🙁 Actual behavior

It reports `error TS1005`:
``` log
R:/working/a.ts:4:27 - error TS1005: ')' expected.

4 const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
                            ~

R:/working/a.ts:4:60 - error TS1005: ',' expected.

4 const a = ver < (MyVer.v1 >= MyVer.v2 ? MyVer.v1 : MyVer.v2)
                                                             ~


Found 2 errors in the same file, starting at: R:/working/a.ts:4
```

### 🙂 Expected behavior

The part between `(` and `)` should be parsed as an operand of `<`.

**Comment by MartinJohns** [comment]
Most likely another side-effect from #47607.
